src/lib/components/ui/FilterBuilder.svelte

```diff
@@ -125,8 +147,54 @@
 		}
 	}
 
+	// Centralized mapping field metadata
+	const mappingFieldConfig: Record<
+		string,
+		{
+			type: string;
+			ui_type: string;
+			options?: Array<{ value: string; label: string }>;
+		}
+	> = {
+		has_mappings: {
+			type: 'boolean',
+			ui_type: 'select',
+			options: [
+				{ value: 'true', label: 'Yes' },
+				{ value: 'false', label: 'No' }
+			]
+		},
+		mapping_status: {
+			type: 'string',
+			ui_type: 'short_text',
```

Comment:
I think this should be a select?

```suggestion
			ui_type: 'select',
```